File: services/auth_service.py
from typing import Any, Dict
from urllib.parse import urljoin
import httpx
from fastapi import HTTPException
from core.settings import settings

async def authenticate(nonce: str, signature: str) -> Dict[str, Any]:
    base_url = settings.AUTH_BASE_URL.rstrip("/") + "/"
    endpoint_path = settings.AUTH_ENDPOINT_PATH.lstrip("/")
    full_url = urljoin(base_url, endpoint_path)

    payload = {
        "nonce": nonce,
        "signature": signature,
        "external": True,
    }
    headers = {
        "Content-Type": "application/json"
    }

    try:
        async with httpx.AsyncClient() as client:
            response = await client.post(full_url, json=payload, headers=headers)
            if response.status_code != 200:
                raise HTTPException(
                    status_code=response.status_code,
                    detail=f"Ошибка аутентификации: {response.text}"
                )
            return response.json()
    except httpx.HTTPStatusError as exc:
        raise HTTPException(
            status_code=exc.response.status_code,
            detail=f"Ошибка HTTP: {exc.response.text}"
        )
    except Exception as e:
        raise HTTPException(
            status_code=500,
            detail=f"Ошибка при выполнении запроса аутентификации: {str(e)}"
        )

File: static/js/auth_service.js
export async function fetchNonce() {
    try {
        const response = await fetch('/get', {
            method: 'POST',
            headers: {
                'Content-Type': 'application/json'
            }
        });
        
        if (!response.ok) {
            throw new Error('Ошибка получения nonce');
        }
        
        const data = await response.json();
        return data.nonce;
    } catch (error) {
        throw error;
    }
}

export async function sign(signature, nonce) {
    try {
        const response = await fetch('/check', {
            method: 'POST',
            headers: {
                'Content-Type': 'application/json'
            },
            body: JSON.stringify({
                signature: signature,
                nonce: nonce
            })
        });
        
        if (!response.ok) {
            throw new Error('Ошибка проверки подписи');
        }
        
        const data = await response.json();
        return data.user_id;
    } catch (error) {
        throw error;
    }
}
